Core/Services/StitchCoreFCMService/Sources/StitchCoreFCMService/Internal/CoreFCMServiceClient.swift

```diff
@@ -0,0 +1,59 @@
+import Foundation
+import MongoSwift
+import StitchCoreSDK
+
+public class CoreFCMServiceClient {
+    private let service: CoreStitchServiceClient
+    
+    public init(withServiceClient serviceClient: CoreStitchServiceClient) {
+        self.service = serviceClient
+    }
+    
+    public func sendMessage(to recipient: String,
+                            withRequest request: FCMSendMessageRequest) throws -> FCMSendMessageResult {
+        return try sendMessageInternal(
+            request: request,
+            targetTypeKey: SendFields.toField.rawValue,
+            targetTypeValue: recipient
+        )
+    }
+    
+    public func sendMessage(toUserIDs userIDs: [String],
+                            withRequest request: FCMSendMessageRequest) throws -> FCMSendMessageResult {
+        return try sendMessageInternal(
+            request: request,
+            targetTypeKey: SendFields.userIDsField.rawValue,
+            targetTypeValue: userIDs
+        )
+    }
+    
+    public func sendMessage(toRegistrationTokens registrationTokens: [String],
+                            withRequest request: FCMSendMessageRequest) throws -> FCMSendMessageResult {
+        return try sendMessageInternal(
+            request: request,
+            targetTypeKey: SendFields.registrationTokensField.rawValue,
+            targetTypeValue: registrationTokens
+        )
+    }
+    
+    private func sendMessageInternal<T: BsonValue>(request: FCMSendMessageRequest,
+                                     targetTypeKey: String,
```

Comment:
[minor] This should be of type `SendFields` and we should take the `rawValue` within this method

Reply:
:+1: